packages/design-system/CONTRIBUTING.md

```diff
@@ -1,53 +1,140 @@
 
-## States
+### Designing the component's API
 
-## Interactions
+The goal is to outline the API, without necessarily coding the component itself.
 
-## Content 
+It can be something like that:
 
-## Usage
 
-## Accessibility
+```tsx
+type Data = {
+	name: string;
+	values: number[];
+	color: string;
+};
 
+type DataSet = Data[];
+
+type ComponentProps = {
+	isActive: boolean;
+	dataset: DataSet;
+}
+
+export function Component(props: ComponentProps) {...}
+
+---
+
+import { Component} from '@talend/design-system'
+import { tokens } from '@talend/design-tokens'
+
+<Component
+	isActive
+	dataset=[{ name: "pouet", values: [1,4,8], color: tokens.coralColorChartPositive }]
+/>
 ```
+
+**Where do I document this API?**
+
+Remember that Storybook page created by your friendly designer team? Once they've set this up, you can expand that component's documentation. That's where the API part comes up.
+
+If they haven't created yet, you can use the [provided template for doc pages](DOCTEMPLATE.md).
+
+Say you're working on a `Dropdown` component. Your friendly designer will have created a `src/components/Dropdown/Dropdown.stories.mdx` file already. Use the `## Usage` section to showcase your API!
+
+**Should I keep this documentation once the component is ready?**
+
+For simple APIs, not really. Your component's stories must be thorough enough to explain its API. For more stateful components or hooks / behaviours (ex: a toast library, a notification center) etc... then yes, absolutely.
+
+### Implementing the component
+
+Your API has been approved? It's time to build.
+
+**Do**
+
+- Name your variables explicitly (no `const v = 'toto'`)
+- Use exclusively design tokens for styles (color, spacing, fonts, border-radiuses, box-shadows, border styles and sizes, breakpoints, transitions, backgrounds, text colors etc...)
+- Write exhaustive explicit stories in TS to showcase actual use cases (in `ComponentName.stories.tsx`) and call those stories in the doc file (in `/ComponentName/ComponentName.stories.mdx`)
+- Write tests with Cypress for your component and its logic if it must have logic (`ComponentName.spec.tsx`)
+- Make sure your component allows for `refs` passing through (`ForwardRef`)
+- Make your component's types match the native HTML element it features (ie: it should able to support data attributes etc...)
+- Make sure your component is exported in the index.ts to be available in the UMD
+
+**Don't**
+
+- Let consumers rely on `className` or `style` or `css` props. The library is there for homogeneity purposes across products. Overrides should be discouraged.
+
+Your component needs at least one positive Codeowner review and one positive review from somebody outside your team (shared library = shared responsibilities). It also need to be approved by the lead designer working on that component (maybe not on Github since all designers don't have access). Once it does, it's up to you to merge and release.
```

Comment:
maybe need to clarify how to have the approval of the designer and how to refer it in the PR

Reply:
Added

```
If the designer working with you on that component has a Github account, request their review. 

If not, reach out on Slack through a public channel (`#ak-ux` or `#ask-designsystem`) and get their approval there.
```